modules/dnn/src/layers/instance_norm_layer.cpp

```diff
@@ -0,0 +1,94 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#include "../precomp.hpp"
+#include <opencv2/dnn/shape_utils.hpp>
+
+namespace cv { namespace dnn {
+
+// https://github.com/onnx/onnx/blob/main/docs/Operators.md#InstanceNormalization
+class InstanceNormLayerImpl CV_FINAL : public InstanceNormLayer {
+public:
+    InstanceNormLayerImpl(const LayerParams &params) {
+        setParamsFrom(params);
+
+        epsilon = params.get<float>("epsilon", 1e-5);
+    }
+
+    virtual bool supportBackend(int backendId) CV_OVERRIDE {
+        return backendId == DNN_BACKEND_OPENCV;
+    }
```

Comment:
Breaks coverage with other backends that initialized in MVN layer (OpenVINO, CUDA).

Reply:
I will try to add them back.

Reply:
OpenVINO and CUDA backend are added.

Reply:
Can you handle OpenCL backend as well? https://github.com/opencv/opencv/blob/ea47cb3ffeedae2da53acbaff8a108cc39a1a4b9/modules/dnn/src/layers/mvn_layer.cpp#L208

Reply:
I can give it a shot.

Reply:
May I ask why returning false in `forward_ocl` in `mvn_layer.cpp` if the input umat type is int16?

https://github.com/opencv/opencv/blob/fa56623458e4e582f16b216448e21bb9e664a9ec/modules/dnn/src/layers/mvn_layer.cpp#L229-L230

Initially I thought it was `CV_16F`, but it is `CV_16S` indeed. So the opencl backend supports int8 quantization as well? But the weirder thing is we don't support int16 quantization right?

Reply:
`CV_16S` is an original data type for OpenCV FP16. Every OpenCL uses `CV_16S` which actually just a short int buffer for float16. See [convertFp16](https://docs.opencv.org/5.x/d2/de8/group__core__array.html#ga9c25d9ef44a2a48ecc3774b30cb80082)

Reply:
Done with adding OpenCL backend (no fp16 though).